x-pack/plugins/security/server/authentication/providers/base.mock.ts

```diff
@@ -4,9 +4,6 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import sinon from 'sinon';
```

Comment:
__note:__ that's what caused 50%+ of the changes in this PR - I finally decided to drop support for `sinon` in auth provider tests as more changes in auth providers is coming (Login Selector) and I don't want to increase technical debt.